### ISSUE TITLE
Why is ksm.h relicensed to GPL 3.0 in this patch?

### ISSUE THREAD

**Issue by abucodonosor** [dolohow/uksm#56]
Hello,

https://github.com/dolohow/uksm/blob/master/v5.x/uksm-5.5.patch#L149-L150 relicenses
the file to GPL 3.0.

Why?.

**Comment by dolohow** [comment]
@naixia Are you able to explain?

**Comment by pfactum** [comment]
I'm not sure @naixia will respond. I haven't seen him online for quite some time, and given his last message here was about him being ill, the outcome might be not the most pleasant.

**Comment by abucodonosor** [comment]
Well, the file cannot be relicensed like this, I assume it is a typo or alike.

Meanwhile, can you just remove that hunk because it is a GPL violation.

**Comment by pfactum** [comment]
OK, I've got a reply from his colleague in Nanjing.

```
He is much better now, but still in a recovery procedure. Cannot work at this time.

He asked me to say "hello and take care" to you, thanks you for so many years' support.
```

**Comment by dolohow** [comment]
> Meanwhile, can you just remove that hunk because it is a GPL violation.

I think GPLv3 is compatible with GPLv2.  Not sure why was relicensed though.

Care to send PR @abucodonosor?

**Comment by abucodonosor** [comment]
It is not a matter of being compatible, you cannot relicense other's work without an agreement of everyone who owns that code. In the Kernel case that cannot be done, ever, it is GPLv2 only. Modules could have a dual GPLv2 / something license but that is about it.

A PR against a patch? Or do you want a PR like a 're-diffing' of the patch against some kernel-version(s)?

**Comment by dolohow** [comment]
Fixed in 5.7.  Anyone care to port to older patches?

And also thanks @abucodonosor for report.

Best
